library/src/layout/page.rs

```diff
@@ -352,6 +357,16 @@ impl PageElem {
             // The padded width of the page's content without margins.
             let pw = frame.width();
 
+            // Make a copy of the original margin in case you need to modify the
+            // margins for a two-sided document.
+            let mut margin = margin;
+
+            // Swap right and left margins if the document is two-sided and the
+            // page number is even.
+            if self.margin(styles).two_sided.unwrap_or(false) && number.get() % 2 == 0 {
```

Comment:
It's unnecessary to fetch the margin property again (you already called `self.margin(styles)` before, and, since the `styles` instance is the same, the returned value shouldn't change). Try keeping the `self.margin(styles)` at lines 308 and 309 in a separate variable from the `.sides` you extract from it. You could do that by e.g. creating a `let margin = self.margin(styles);` variable and then using `let margin_sides = margin.sides.map(...)` elsewhere, or something like that.

Reply:
Done.